chain/validation/applier.go

```diff
@@ -42,6 +41,10 @@ func (a *Applier) ApplyMessage(eCtx *vchain.ExecutionContext, state vstate.Wrapp
 		return vchain.MessageReceipt{}, err
 	}
 
+	// FIXME this is pretty hacky (it works), flushing the lotusVM fails becasue the root of lotusVM.StateTree is not
+	// its buffered blockstore.
+	st.StateTree = lotusVM.StateTree().(*lstate.StateTree)
+
```

Comment:
This won't actually persist the blocks to the store. I've made some changes here that I think will get closer, but the vm.Flush() is failing.

Reply:
From what I can tell it fails for 2 reasons:
1. The chain-validation package generates actor code CID's differently than lotus :
   - https://github.com/filecoin-project/chain-validation/blob/master/pkg/state/actors.go#L26
   - https://github.com/filecoin-project/lotus/blob/master/chain/actors/actors.go#L40

This can be solved by implementing an ActorCodeID enum that maps to each implementation as we discussed out of band.

2. The EmptyObjectCID that actors have their head set to is not placed in the blockstore when we call NewState: https://github.com/filecoin-project/lotus/blob/master/chain/vm/mkactor.go#L21, thus it is not found when we Flush.

Currently solving this by mimicking the behaviour of the linked init method inside `NewState`.